### ISSUE TITLE
General comment field improvements

### ISSUE THREAD

**Issue by kwa20** [hzi-braunschweig/SORMAS-Project#3953]
### Feature Description
General comment field (wherever applicable)
### Problem Description
Thus far, the general comment field only provides an option to freely enter comments or anything further in one field. Often enough, users use this field to enter a lot of additional data. However, it cannot be inferred, who entered data or when it was entered.
### Proposed Change
To optimize the comment field for internal communication of users, change this section to allow stacking of comments. A comment should be added by pressing "new comment". After saving the comment, user name, user role and creation date and time should be displayed in a comment header. The individual comment fields should be able to be minimized (hidden) and maximized. This is important to maintain clarity and not having to scroll for very long to reach the discard and save buttons on the form. When a comment is minimized, merely the header should be visible with the arrow implying that the field can be maximized. When a user comes back to his comment, they should have an option to edit it. The time of the edit should then update the time stamp and the edit symbol should be added to show that the comment was edited.

### Additional Information
<img width="605" alt="generalComment2" src="https://user-images.githubusercontent.com/70317594/104125516-8b581d80-5357-11eb-801a-3ef262ff8a02.PNG">

**Comment by kwa20** [comment]
Will likely influence #3936 and #3898

**Comment by Marko-ilmkreis** [comment]
That would be very good improvement of this field. Thanks for your work on the weekend!

**Comment by kwa20** [comment]
This issue should also apply to the general comment field that would be added to persons #3936

**Comment by markusmann-vg** [comment]
@JaquM @kwa20 
We will do an internal refinement for this

Examples for our internal refinement:
- Data modeling
- Number of comments to show, before they can be collapsed
- What about editing, deleting comments
- How do we migrate the old comments

**Comment by StefanKock** [comment]
This is likely a complete new feature

**Comment by markusmann-vg** [comment]
to be considered: https://github.com/hzi-braunschweig/SORMAS-Project/issues/3936

**Comment by bernardsilenou** [comment]
This is similar to whet we have for event actions. It would be good to use the same design and logic in both sections

**Comment by kwa20** [comment]
@bernardsilenou I actually don't think it should work like event actions since actions are more related to organizing larger endeavours and this section is mainly for internal commentary on department level. Therefore, users should not have to navigate to separate tabs or go through multiple creation windows.

**Comment by bernardsilenou** [comment]
@kwa20 
- I was not referring to adding another tab but the internal documentation of the comment itself.
- I just have a second and validate that this is not the same as actions since history of each change is not documented.
- This looks more like a chat function within the comment section. I understand now, thanks

**Comment by maxiheyner** [comment]
Issue tracker comment

**Comment by MateStrysewske** [comment]
Some quick notes based on a discussion we just had:

* The mockup above is basically what we need
* Users should be able to edit and (soft-)delete their own comments, or all comments if they have the user right to do so
* Edit information needs to be shown (who edited last and when)
* Always display the 5 most recent entries, or the 4 most recent entries and one entry that has been pinned (highlight the pinned entry); automatically collapse others
* Add functionality to "expand all other entries" or "collapse all other entries"
* Adding a new comment should not be done by clicking on a "New comment" button; instead, there should be an empty comment on top of the list with a discard & save button that the user can just type into; if they click on save, the comment is added to the list

**Comment by MateStrysewske** [comment]
@kwa20 What about migration? We should get rid of the "additionalDetails" fields for cases, contacts and persons because those are currently serving as comment fields. However, those fields don't have an associated "creating user", so if we migrate them to the new comments, these comments wouldn't have a user associated with them. We could allow this and introduce a user right to edit unowned comments, but that would be kind of a hacky workaround only for the migration. An alternative could be to automatically assign these comments to a random admin user. What do you suggest here?

**Comment by kwa20** [comment]
@MateStrysewske We could use the reporting user as the user to be the 'creating user'. The downside is that this might not necessarily be true and I'd therefore discard this option. I also agree that creating an extra user right for this would not be that nice either. However, considering you're previous comment:

> Users should be able to edit and (soft-)delete their own comments, or all comments if they have the user right to do so

Would this right not also make respective users able to edit unowned comments? Otherwise, what if we make unowned comments available for edits to everyone (which is pretty much how the general comment field worked before), but assign/associate the first user that changes the unowned comment after initial migration.

**Comment by MateStrysewske** [comment]
Yeah, good point - I totally missed that. Doesn't seem to be a problem then :)

**Comment by Marko-ilmkreis** [comment]
@MateStrysewske @kwa20 I ant to point out that there must be also a handling for comments left by users which are leaving the health office. This could be the scouts from RKI, soldiers or just an employee with a new job.
I recommend that if the user of the comment is deactivated the comment it editable by any user, but it mus be shown who did the last change. Other option would be to make it editable for admins only.

**Comment by bansaj** [comment]
Can we replicate the comment to other modules like events, etc, or will it be specifically added the above-mentioned models only?

**Comment by JaquM-HZI** [comment]
We could have it for every general comment field and it should be possible to add a general comment field for events @bansaj
